langchain4j-azure-open-ai/src/main/java/dev/langchain4j/model/azure/AzureOpenAiChatModel.java

```diff
@@ -228,28 +235,54 @@ private AzureOpenAiChatModel(String deploymentName,
         this.dataSources = dataSources;
         this.enhancements = enhancements;
         this.seed = seed;
-        this.responseFormat = responseFormat;
+        this.chatCompletionsResponseFormat = chatCompletionsResponseFormat;
         this.listeners = listeners == null ? emptyList() : new ArrayList<>(listeners);
     }
 
+    @Override
+    public Set<Capability> supportedCapabilities() {
+        Set<Capability> capabilities = new HashSet<>();
+        if (chatCompletionsResponseFormat != null && chatCompletionsResponseFormat instanceof ChatCompletionsJsonSchemaResponseFormat) {
```

Comment:
I was thinking about this logic lately and I think that it would be clearer if user enables it like this:

```java
XyzChatModel().builder()
    ...
    .supportedCapabilities(List.of(RESPONSE_FORMAT_JSON_SCHEMA, ...))
    .build();
```

On the other hand, when user sets the response format (example below), it sohuld be applied to all the requests to the LLM, but it should not drive the logic of the `supportedCapabilities()` method.
```java
XyzChatModel().builder()
    ...
    .responseFormat(...)
    .build();
```

WDYT?

Reply:
I get it!
As I put in https://github.com/langchain4j/langchain4j/pull/1982#issuecomment-2532127682 I preferred to have 3 options in the `ResponseFormat`,  but your solution also works and has the advantage of not breaking the existing OpenAI and Gemini code.
I've just pushed a commit that supports your solution, can you have a look? Looking at `AzureOpenAiAiServicesWithJsonSchemaIT`, I believe it's OK for end users.

Reply:
Just in case, `supportedCapabilities(List.of(RESPONSE_FORMAT_JSON_SCHEMA, ...))` is required only when using AI Services (because AI Services logic sohuld know in what format to generate schema: append to the prompt or generate `JsonSchema` for models that support this feature). It is not required to be set when using `AzureOpenAiChatModel` directly.